main.py

```diff
@@ -66,57 +85,59 @@ def create_overlays(ground, ego):
+key_indicator_pressed = LandingGameObject(
+    img_key_indicator_pressed,
     Vec2d(CommonConstants.WINDOW_WIDTH - 50, CommonConstants.WINDOW_HEIGHT - 50),
 )
+img_key_indicator_down = create_pg_surface_from_color_and_size(colors_dict["cyan"], (20, 20))
+
+key_indicator_down = LandingGameObject(
+    img_key_indicator_down,
+    Vec2d(CommonConstants.WINDOW_WIDTH - 100, CommonConstants.WINDOW_HEIGHT - 50),
+)
+
 ego = Rocket(img_ego, rocket_pos, rocket_mass)
 ground = LandingGameObject(img_ground, ground_position)
 overlays = create_overlays(ground, ego)
 
 # predefine actions on key: keypress-states connected to actions that will be performed if state is given
-color_key_indicator_blue = lambda: key_indicator.set_color(colors_dict["blue"])
-color_key_indicator_cyan = lambda: key_indicator.set_color(colors_dict["cyan"])
+color_key_indicator_blue_pressed = lambda: key_indicator_pressed.set_color(colors_dict["blue"])
+color_key_indicator_cyan_pressed = lambda: key_indicator_pressed.set_color(colors_dict["cyan"])
+
+color_key_indicator_blue_down = lambda: key_indicator_down.set_color(colors_dict["blue"])
+color_key_indicator_cyan_down = lambda: key_indicator_down.set_color(colors_dict["cyan"])
 
 act_change_box_color_on_spacebar_pressed = LandingGameActionOnKey(
-    PygameKeyState(pygame.K_SPACE, True), color_key_indicator_blue
+    PygameKeyState(pygame.K_SPACE, True), color_key_indicator_blue_pressed
 )
 act_change_box_color_on_spacebar_not_pressed = LandingGameActionOnKey(
-    PygameKeyState(pygame.K_SPACE, False), color_key_indicator_cyan
+    PygameKeyState(pygame.K_SPACE, False), color_key_indicator_cyan_pressed
+)
+
+act_change_box_color_on_spacebar_down = LandingGameActionOnKey(
+    PygameKeyState(pygame.K_SPACE, True), color_key_indicator_blue_down
 )
 
 
 obj_list = pygame.sprite.Group()
-obj_list.add(key_indicator)
+obj_list.add(key_indicator_pressed)
+obj_list.add(key_indicator_down)
 obj_list.add(ego)
 obj_list.add(ground)
 for overlay in overlays:
     obj_list.add(overlay)
 
 # bundle all keystate -> action correlations into one list
-actions_on_key = [
+actions_on_key_pressed = [
     act_change_box_color_on_spacebar_pressed,
     act_change_box_color_on_spacebar_not_pressed,
 ]
+actions_on_key_down = [act_change_box_color_on_spacebar_down]
 
 while True:
-    for event in pygame.event.get():
-        if event.type == pygame.locals.QUIT:
-            pygame.quit()
-            sys.exit()
-        else:
-            # check all key states and automatically perform all callbacks
-            key_press_state = pygame.key.get_pressed()
-            for action in actions_on_key:
-                trigger_action_given: bool = (
-                    action.trigger_key.pressed
-                    == key_press_state[action.trigger_key.key_identifier]
-                )
-                if trigger_action_given:
-                    action.execute_action()
-
+    process_events(actions_on_key_pressed, actions_on_key_down)
```

Comment:
Nice structure! Since this function only handles key press events, maybe call it sth like process keyboard events to describe its purpose even more accurately.

Reply:
Good point!